mono/mini/simd-intrinsics-netcore.c

```diff
@@ -686,9 +706,6 @@ emit_x86_intrinsics (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature
 		}
 	}
 	if (!strcmp (class_name, "Bmi2") || (!strcmp (class_name, "X64") && cmethod->klass->nested_in && !strcmp (m_class_get_name (cmethod->klass->nested_in), "Bmi2"))) {
-		// We only support the subset used by corelib
-		if (m_class_get_image (cfg->method->klass) != mono_get_corlib ())
-			return NULL;
```

Comment:
I don't think you can safely remove this until `SN_MultiplyNoFlags` is implemented. `IsSupported` will return `true` but the actual method will end up with throwing `PlatformNotSupportedException` (or similar).

Reply:
yeah... but it unlocks Bmi2 for users, MultiplyNoFlags is currently problematic in CoreCLR: https://github.com/dotnet/coreclr/issues/21899

Reply:
I do realize that but even an unoptimized fallback would be better option. The linked issue shows that there are actually people trying to use it. For those people the situation with `Bmi2.IsSupported == true` would suddenly start throwing exceptions in perfectly legitimate code.

Reply:
Yes, that check should only be removed if all the methods are supported.

Reply:
@filipnavara @vargaz ok